src/core/feed.factory.ts

```diff
@@ -84,22 +84,29 @@ export class FeedFactory {
     return new BlockedUsersFeed(this.client);
   }
 
-  public directInbox(): DirectInboxFeed {
-    return new DirectInboxFeed(this.client);
+  public directInbox(options: { limit: number; thread_message_limit: number }): DirectInboxFeed {
+    const feed = new DirectInboxFeed(this.client);
+    feed.limit = options.limit;
+    feed.thread_message_limit = options?.thread_message_limit;
```

Comment:
You may want to use `plainToClassFromExist`

Also: `options` is not undefined, so `options?.` is unnecessary.